data_model/src/block.rs

```diff
@@ -280,16 +280,18 @@ impl SignedBlock {
 
     /// Creates genesis block signed with genesis private key (and not signed by any peer)
     pub fn genesis(
-        genesis_transaction: SignedTransaction,
+        genesis_transactions: Vec<SignedTransaction>,
```

Comment:
[Parse, don’t validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)

Maybe it would be good to define a stricter type for `GenesisTransactions`, which will verify on construction/deserialization that the first TX has only one `Upgrade` ISI?

```rs
// (De-)serializes as `Vec<SignedTransaction>` with validation
struct GenesisTransactions {
  upgrade: SignedTransaction,
  rest: Vec<SignedTransaction>
}

impl GenesisTransactions {
  fn new(upgrade: SignedTransaction, rest: Vec<SignedTransaction>) -> Result<Self, SomeError> {
    // validate that `upgrade` is appropriate
  }

  fn get_upgrade(&self) -> Upgrade {
    // can `unwrap` `self.upgrade` safely thanks to validation on creation
  }
}
```

Reply:
Let's make it a separate issue and merge as is